packages/web-api-scan-job-manager/src/worker/worker.ts

```diff
@@ -76,18 +81,58 @@ export class Worker {
             if (moment().toDate() >= this.restartAfterTime) {
                 this.logger.logInfo(`Performing scheduled termination after ${this.jobManagerConfig.maxWallClockTimeInHours} hours.`);
                 await this.waitForChildTasks();
+                await this.completeTerminatedTasks();
+
                 break;
             }
 
             await this.system.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
         }
     }
 
+    private async completeTerminatedTasks(): Promise<void> {
+        const failedTasks = await this.batch.getFailedTasks(this.batchConfig.jobId);
+        await Promise.all(
+            failedTasks.map(async failedTask => {
+                // tslint:disable-next-line: no-unsafe-any
+                const taskArguments = JSON.parse(failedTask.taskArguments) as TaskArguments;
+                if (taskArguments !== undefined && taskArguments.id !== undefined) {
+                    let error = `Task was terminated unexpectedly. Exit code: ${failedTask.exitCode}`;
+                    error =
+                        failedTask.failureInfo !== undefined
+                            ? `${error}, Error category: ${failedTask.failureInfo.category}, Error message: ${
+                                  failedTask.failureInfo.message
+                              }`
+                            : error;
+
+                    let pageScanResult = await this.onDemandPageScanRunResultProvider.readScanRun(taskArguments.id);
+                    if (pageScanResult !== undefined) {
+                        if (pageScanResult.run.state !== 'failed') {
+                            pageScanResult.run = {
+                                state: 'failed',
+                                timestamp: failedTask.timestamp.toJSON(),
+                                error,
+                            };
+                            pageScanResult = await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+                        }
```

Comment:
can you log here the failure info from batch here? incase we have 2 different failures (scan failure message is different from task failure)

Reply:
batch error logged below
`                    this.logger.logError(error, { taskProperties: JSON.stringify(failedTask) });
`